dist_run.py

```diff
@@ -418,9 +418,7 @@ def get_example_ins_outs(seqlen: int) -> Tuple[torch.Tensor, torch.Tensor]:
 
     # Decode token id into string and print it
     def decode_in_flight(token):
-        # Make a 2D tensor with ids on row dimension
-        unsqueezed = torch.unsqueeze(token, 1)
-        token_str = tokenizer.decode(unsqueezed.tolist())
+        token_str = tokenizer.decode(token.tolist())
```

Comment:
I find that this does not work as-is. 
However, adding in a one liner:
~~~
token = token.squeeze(1)
~~~
And now it works on both llama2 and llama3.

Reply:
Ah, makes sense. Tokenizer difference :)

Reply:
It seems I cannot unconditionally squeeze the tensor. 
If I do, some tokenizers will output:
`responses ====>>>>  is Christmasiving`
istead of
`responses ====>>>>  ['is', 'Christmas', 'iving']`

Reply:
I am:
`using tokenizer = sentencepiece.SentencePieceProcessor`

Reply:
Okay, I am adding an `if` there:
```
    # `token` is a tensor of shape (batch_size, 1).
    # For TiktokenTokenizer, we need to squeeze it to 1D.
    # For SentencePieceProcessor, we don't.
    if isinstance(tokenizer, TiktokenTokenizer):
        token = torch.squeeze(token, dim=1)
    token_str = tokenizer.decode(token.tolist())
```